editor/editor_help.cpp

```diff
@@ -2415,14 +2419,15 @@ void EditorHelp::_gen_doc_thread(void *p_udata) {
 	}
 }
 
+void EditorHelp::_gen_extensions_docs() {
```

Comment:
I think this can be a public method (called `generate_extension_docs`), and I think to fully support extensions this needs to be called whenever the GDExtension manager hot reloads any library. IIUC, this right now only loads extension docs when loading the cache or generating it anew.

Reply:
In @Riteo's PR to allow GDExtensions to add fuller documentation, he's got a separate `DocTools *` just for the extension documentation in order to keep track of which is from extensions:

https://github.com/godotengine/godot/pull/83747/files#diff-7f898e9b24837a348e2209c5a7e5d2b686d91580787069a6bab6ea78a5d38dba

Perhaps that approach could work here?

(May not actually make sense, I don't know the documentation system that well :-))

Reply:
I don't think this would be necessary for this particular purpose (and besides, when we generate everything anew, we don't differentiate between sources of each class). ClassDB already has the API type stored, we can just read it and add it to the documentation entry.